### ISSUE TITLE
Add 'update' functionality to the dita install subcommand

### ISSUE THREAD

**Issue by super-structure** [dita-ot/dita-ot#4325]
## Description
It could be useful to have an `update` option use with the plugin registry. This would simply function similar to PIP or NPM, in which an additional optional (or different subcommand; update instead of install) is used to quickly update. 

## Possible Solution
This might just be a synonym command for the `--force` option, but perhaps with some notes regarding the current and latest versions and a confirmation. E.g., Are you sure you want to update the com.elovirta.pdf plugin from 0.7.0 to the latest available version (0.7.1)? [y/N]

**Note:** The `--force` option also seems to occasionally run into issues being unable to delete files within a plugin. Not sure if this is an issue worthy of a bug or not. E.g. 
```Error: [java.io.IOException: Unable to delete file: D:\DITA\DITA-OT_4.1.2_Dev\bin\..\plugins\com.elovirta.pdf\lib]``` 

- a `--info` option here could be used to show the current installed version of all registry plugins along with the latest available (com.elovirta.pdf@0.7.0 -> 0.7.1)
- a `--all` option could be used to update all registry plugins to the latest available version.
- a `--version` option could take a specific version to download and install, rather than just the latest available (would not work with an `--all` option, I would think)

## Potential Alternatives
This option could also be extended to a subcommand for the DITA-OT itself, where `dita update` would download and install the latest (or specified) version of the DITA-OT.

Also, a `--registry` option could be added to the `dita plugins` command to show only those plugins which are available on the registry.

## Additional Context
The benefit of having a specific update command would bring the dita CLI in-line with many modern package managers. While I realize the plugin registry isn't really trying to be a package manager, it seems like a minor yet useful improvement for those used to this style of CLI.

**Comment by jelovirt** [comment]
These are good suggestion, but there are some problems that come with them.

The "unable to delete" is a result of OT trying to delete a JAR file it's currently running. A fix would be to run install and uninstall subcommands with a classpath that only contains the core libraries, none of the plugin libraries. For Bash CLI this is a simple fix, but for Windows the problem is that none of the main developers has a Windows machine. Maybe a fix for Windows could be left for the community to contribute.

**Comment by infotexture** [comment]
> we'd need to be a lot more careful with the version dependency metadata for built-in plug-ins

Indeed. The safe default would be to only bump plug-ins to newer versions that are explicitly declared compatible with the installed toolkit version, and maybe provide something like the `--force` option install to override that default and also install newer plug-ins that aren't officially compatible with the installed version.

**Comment by infotexture** [comment]
@jelovirt While you're thinking about safeguarding `uninstall`, we might want to consider reviewing plug-in dependencies as part of the process, so if someone tries to remove a plug-in that another depends on, the action fails with a warning that explains why.

Here's an example of how Homebrew handles this:

```
❯ brew uninstall ruby
Error: Refusing to uninstall /opt/homebrew/Cellar/ruby/3.2.2_1
because it is required by gem-bundler, which is currently installed.
You can override this and force removal with:
  brew uninstall --ignore-dependencies ruby
```

We can treat this as a separate issue of course, just wanted to mention it here in case it makes sense to combine them.

**Comment by super-structure** [comment]
@jelovirt It seems obvious now that you say that's why it could it be removed; not sure why that hadn't occurred to other than having never tried that command before. As a Windows user, I'll be glad to work on (at least) the ability to kill a running .jar file. I think it's also a pretty straightforward thing to do via a command based on a [cursory search in StackOverflow](https://stackoverflow.com/a/13025928).
